cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/group/GcpInstanceGroupResourceBuilder.java

```diff
@@ -0,0 +1,154 @@
+import com.google.api.services.compute.model.InstanceGroupsListInstancesRequest;
+import com.google.api.services.compute.model.InstanceGroupsRemoveInstancesRequest;
+import com.google.api.services.compute.model.InstanceReference;
+import com.google.api.services.compute.model.InstanceWithNamedPorts;
+import com.google.api.services.compute.model.Operation;
+import com.google.common.collect.Sets;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.gcp.GcpResourceException;
+import com.sequenceiq.cloudbreak.cloud.gcp.context.GcpContext;
+import com.sequenceiq.cloudbreak.cloud.model.CloudInstance;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResourceStatus;
+import com.sequenceiq.cloudbreak.cloud.model.Group;
+import com.sequenceiq.cloudbreak.cloud.model.Location;
+import com.sequenceiq.cloudbreak.cloud.model.Network;
+import com.sequenceiq.cloudbreak.cloud.model.Security;
+import com.sequenceiq.common.api.type.ResourceType;
+
+/**
+ * The Group Resource Builder that is responsible for the GCP API calls to manage an instance group
+ * This currently only defines unmanaged instance groups
+ * In GCP an instance group is a collection of Compute Instances in the same Zone and subnet
+ * An unmanaged instace group is only used to be applied as a target of a load balancer backend
+ * An instance shoup be added to at most 1 load balanced instance group
+ * This creates a Instance Group for every group defined in a stack.
+ */
+
+@Service
+public class GcpInstanceGroupResourceBuilder extends AbstractGcpGroupBuilder {
+
+    private static final int ORDER = 1;
+
+    @Override
+    public CloudResource create(GcpContext context, AuthenticatedContext auth, Group group, Network network) {
+        String resourceName = getResourceNameService().resourceName(resourceType(), context.getName(), group.getName());
+        return createNamedResource(resourceType(), resourceName);
+    }
+
+    @Override
+    public CloudResource build(GcpContext context,
+            AuthenticatedContext auth, Group group,
+            Network network, Security security, CloudResource resource) throws Exception {
+
+        Insert insert = context.getCompute().instanceGroups().insert(context.getProjectId(),
+                context.getLocation().getAvailabilityZone().value(), new InstanceGroup().setName(resource.getName()));
+
+        return doOperationalRequest(resource, insert);
+    }
+
+    @Override
+    public CloudResourceStatus update(GcpContext context, AuthenticatedContext auth, Group group, Network network, Security security, CloudResource resource) {
+        return null;
+    }
+
+    /*
+     * compute difference between current Group and GCP Instance Group, remove any no longer needed then add new
+     */
+    protected void updateInstanceList(GcpContext context, AuthenticatedContext auth,
```

Comment:
**suggestion:** I think this can actually be package-private as well, since the test class is in the same package.